sql/sql_table.cc

```diff
@@ -8473,11 +8473,21 @@ mysql_prepare_alter_table(THD *thd, TABLE *table,
     }
   }
 
-  if (table->versioned() && !(alter_info->flags & ALTER_DROP_SYSTEM_VERSIONING) &&
-      new_create_list.elements == VERSIONING_FIELDS)
+  if (table->versioned())
   {
-    my_error(ER_VERS_TABLE_MUST_HAVE_COLUMNS, MYF(0), table->s->table_name.str);
-    goto err;
+    uint versioned_fields= 0;
+    field_it.rewind();
+    while (Create_field *f= field_it++)
```

Comment:
Maybe early return the loop if we already counted more than VERSIONING_FIELDS?

Reply:
It will bring unnecessary complications. In particular, the value of `versioned_fields` will be inconsistent after the loop. This can confuse somebody else later.